backends/updates/macosx/macosx-updates.mm

```diff
@@ -74,11 +75,13 @@
 	// Finally give up our references to the objects
 	[menuItem release];
 
-	// Enable automatic update checking once a day (alternatively use
-	// checkForUpdates() here to check for updates on every startup)
-	// TODO: Should be removed when an update settings gui is implemented
-	setAutomaticallyChecksForUpdates(kUpdateStateEnabled);
-	setUpdateCheckInterval(kUpdateIntervalOneDay);
+	if (!ConfMan.hasKey("updates_check")
+			|| ConfMan.getInt("updates_check") == Common::UpdateManager::kUpdateIntervalNotSupported) {
+		setAutomaticallyChecksForUpdates(kUpdateStateDisabled);
+	} else {
+		setAutomaticallyChecksForUpdates(kUpdateStateEnabled);
+		setUpdateCheckInterval(ConfMan.getInt("updates_check"));
```

Comment:
Good point. I just implemented normalizing while setting via API and in the GUI. When incorrect value is provided I fall back to one day.

Currently Sparkle itself restricts internally updates to 3600 seconds, e.g. once per hour.